### PR TITLE
Workload updater decrease test time and complexity

### PR DESCRIPTION
**Pull request by xpivarc**
### What this PR does
This series of commits gradually remove complexity of workload updater tests to the point where we reduce the runtime from ~100 seconds down to ~200ms.

Fixes #

Links to places where the discussion took place: 

### Special notes for your reviewer
I recommend to review commit by commit.

### Release note

```release-note
NONE
```

### CODE REVIEW

File: pkg/virt-controller/watch/workload-updater/workload-updater_test.go [hunk 1]
```diff
@@ -427,8 +396,8 @@ var _ = Describe("Workload Updater", func() {
 			controller.Execute()
 			Expect(recorder.Events).To(BeEmpty())
 
-			// sleep to account for batch interval
-			time.Sleep(3 * time.Second)
+			// Shift time for batch interval
+			controller.lastDeletionBatch = time.Now().Add(-3 * time.Second)
```
Comment (0xFelix) [inline pkg/virt-controller/watch/workload-updater/workload-updater_test.go:400]: Great catch :) Getting rid of `time.Sleep` looks like an awesome improvement.

**Review by 0xFelix — APPROVED** [review]
/lgtm

File: pkg/virt-controller/watch/workload-updater/workload-updater_test.go (hunk 1 above)
Comment (xpivarc) [reply pkg/virt-controller/watch/workload-updater/workload-updater_test.go:400]: Funny enough it was only slight impact.

**Comment by xpivarc** [comment]
/cc @jean-edouard 
Can you have a look please?

File: pkg/virt-controller/watch/workload-updater/workload-updater.go [hunk 2]
```diff
@@ -65,17 +65,19 @@ const defaultBatchDeletionCount = 10
...
-	vmiInformer           cache.SharedIndexInformer
-	podInformer           cache.SharedIndexInformer
-	migrationInformer     cache.SharedIndexInformer
+	vmiStore              cache.Store
+	podIndexer            cache.Indexer
+	migrationStore        cache.Store
 	recorder              record.EventRecorder
 	migrationExpectations *controller.UIDTrackingControllerExpectations
-	kubeVirtInformer      cache.SharedIndexInformer
+	kubeVirtInformer      cache.Store
```
[the remark this answers is not in the document]
Comment (xpivarc) [reply pkg/virt-controller/watch/workload-updater/workload-updater.go]: Done

[the remark this answers is not in the document]
Comment (jean-edouard) [reply pkg/virt-controller/watch/workload-updater/workload-updater.go]: /unhold

File: pkg/virt-controller/watch/workload-updater/workload-updater_test.go [hunk 3]
```diff
@@ -577,7 +573,7 @@ func newKubeVirt(expectedNumOutdated int) *v1.KubeVirt {
 	}
 }
 
-func newVirtualMachine(name string, isMigratable bool, image string, vmiIndexer cache.Indexer, podStore cache.Store) *v1.VirtualMachineInstance {
+func newVirtualMachine(name string, isMigratable bool, image string, vmiIndexer cache.Store, podStore cache.Store) *v1.VirtualMachineInstance {
```
Comment (fossedihelm) [inline pkg/virt-controller/watch/workload-updater/workload-updater_test.go:549]: nit: `podStore cache.Store` can be `podIndexer cache.Indexer`

**Review by fossedihelm — COMMENTED** [review]
Looks good! Thank you @xpivarc for this great improvement!
Just one tiny nit. Feel free to unhold
/lgtm
/hold

File: pkg/virt-controller/watch/workload-updater/workload-updater_test.go (hunk 3 above)
Comment (xpivarc) [reply pkg/virt-controller/watch/workload-updater/workload-updater_test.go:549]: will do

Comment (fossedihelm) [reply pkg/virt-controller/watch/workload-updater/workload-updater_test.go:549]: Yeah, I thought the same after a while.

**Pull request merged**